src/systemTest/java/com/flightstats/hub/testcase/WebhookLifecycleTest.java

```diff
@@ -0,0 +1,198 @@
+package com.flightstats.hub.testcase;
+
+import com.flightstats.hub.BaseTest;
+import com.flightstats.hub.callback.CallbackServer;
+import com.flightstats.hub.client.CallbackResourceClient;
+import com.flightstats.hub.client.ChannelItemResourceClient;
+import com.flightstats.hub.client.ChannelResourceClient;
+import com.flightstats.hub.client.WebhookResourceClient;
+import com.flightstats.hub.model.Channel;
+import com.flightstats.hub.model.ChannelItem;
+import com.flightstats.hub.model.Webhook;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import retrofit2.Call;
+import retrofit2.Response;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@Slf4j
+public class WebhookLifecycleTest extends BaseTest {
+
+    private static final String EMPTY_STRING = "";
+
+    private ChannelItemResourceClient channelItemResourceClient;
+    private ChannelResourceClient channelResourceClient;
+    private WebhookResourceClient webhookResourceClient;
+    private CallbackResourceClient callbackResourceClient;
+    private CallbackServer callbackServer;
+
+    private String channelName;
+    private String webhookName;
+
+    private List<String> channelItemsPosted;
+
+    @Before
+    public void setup() {
+
+        this.channelItemResourceClient = getHttpClient(ChannelItemResourceClient.class);
+        this.channelResourceClient = getHttpClient(ChannelResourceClient.class);
+        this.webhookResourceClient = getHttpClient(WebhookResourceClient.class);
+        this.callbackResourceClient = getHttpClient(CallbackResourceClient.class);
```

Comment:
Would it make sense to use a different retrofit client builder for the callback client, since it seems like it'll have a different base url? Trips created a whole Retrofit client builder to make that a little easier (it's in bitbucket under PDX-TRIPS -> trips-common -> RetrofitClientBuilder.java), though I think it's for retrofit v1, so it'd need tweaking.